packages/components/src/InputFile/InputFile.tsx

```diff
@@ -123,6 +123,11 @@ interface InputFileProps {
    */
   readonly allowMultiple?: boolean;
 
+  /**
+   * Maximum number of files that can be uploaded via the dropzone.
+   */
+  readonly maxFiles?: number;
```

Comment:
I think it would be worth adding a note about the scenario described here
<img width="825" alt="image" src="https://github.com/user-attachments/assets/e4711618-494f-4083-ba8a-cc0e02e74fd3">

Since people are likely going to set it and forget it

Reply:
Another thought could be to do some typescript stuff with `XOR` to require a numberOfFiles prop to be passed to the component to shift this into our component so we can figure out the number of remaining files that can be uploaded